source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,352 @@
+group rules attributes.
+
+All of those attributes are then stored in the ACL external_ids column -
+along with the *description* and *security group id* attributes.
+
+The newly created ACL is then stored in the ovn-nb database, and its
+data is afterwards translated back into the networking-api syntax.
+
+The security group rules also feature the resource timestamps extension,
+but the revision number will always be *one*, since we do not allow a
+rule to be updated.
+
+![add-rule](../../../../images/features/network/security-groups/create_rules.png)
+
+### Attach a VM to an external network.
+
+The ovirt-provider-ovn project already implements the networking-api
+port concept, *but* does not allow it to bind to security groups.
+
+After the API is updated, the cms will be able to specify which security
+groups will be bound to the port.
+
+The ovirt-provider-ovn will have to update the port group data, adding
+the new port to the list of ports referenced by that port group - for
+every networking-api security group in the list.
+
+![add-port](../../../../images/features/network/security-groups/add_ports.png)
+
+## Provider start-up security groups
+
+### drop-all-ip
+
+To achieve the intended deny-all behavior, a port group will be created
+when the ovirt-provider-ovn starts.
+
+It will feature a single ACL, with the following data:
+* priority: DROP_PRIORITY
+* action: drop
+* direction: from-lport
+* match: inport == @<port_group> && ip
+
+Depending on the chosen [security group feature activation](#activating-the-feature) mechanism,
+the all newly created ports will be added to this group *or* only the
+ports featuring port-security activated.
+
+### allow-dhcp
+
+To allow a VM to get an IP address from the subnet on top the network
+to which its port is attached, and assuming all traffic is being blocked
+, there is a need to white-list the DHCP traffic.
+
+This will only be done *if* the VM's port is attached to a network
+**having** a defined subnet.
+
+It will feature two ACLs (one per each direction), with the following
+data:
++ ingress rule
+    * priority: ALLOW_PRIORITY
+    * action: allow
+    * direction: from-lport
+    * match: inport == @<port_group> && ip4 && ip4.dst == {255.255.255.255, <subnet_cidr>} && udp && udp.src == 68 && udp.dst == 67
++ egress rule
+    * priority: ALLOW_PRIORITY
+    * action: allow
+    * direction: to-lport
+    * match: outport == @<port_group> && ip4 && ip4.src == <subnet_cidr> && udp && udp.src == 67 && udp.dst == 68
+
+***Note:*** the '@' sign is a port group tag understood by OVN
+
+## Networking API extensions implemented / not implemented
+
+Implemented:
+* [Resource timestamps](https://developer.openstack.org/api-ref/network/v2/#id367)
+
+Not implemented:
+* [Tag](https://developer.openstack.org/api-ref/network/v2/#id381)
+
+Not sure:
+* [Allowed address pairs](https://developer.openstack.org/api-ref/network/v2/#allowed-address-pairs) networking API extension.
+
+## Testing
+
+TODO
```

Comment:
the API could be tested by using an openstack client, e.g. ansible os_security_group or the command line client.
I wonder if there is a way using less effort to check the functionality?

Reply:
I don't think so; ansible looks to be *the* simplest possible way of interacting with this feature.

Having said that, we need to make sure our docs specify very well which subset of the API we support - I think everything except **tags** should be handled.